### PR TITLE
 getUser and Update group

### PR DESCRIPTION
**Pull request by Salman-kutty**

### CODE REVIEW

File: controllers/userCtrl.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import { Request, Response } from 'express'
+
+import { PrismaClient } from '@prisma/client'
+
+const prisma = new PrismaClient();
+
+export const getAllUser = async (req: Request, res: Response) => {
+    try {
+        const obj= filtering(req.body);
```
Comment (rishibroto-Tectoro) [inline controllers/userCtrl.ts:9]: Not required

File: controllers/userCtrl.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        console.log("Object --> ",obj)
+        const data = await prisma.user.findMany({skip:(req.body.skip -1) * req.body.take ,take:req.body.take , where: obj,include:{groupOwned:true} });
```
Comment (rishibroto-Tectoro) [inline controllers/userCtrl.ts:11]: Add a where clause instead of calling filtering function

Comment (rishibroto-Tectoro) [inline controllers/userCtrl.ts:11]: Is it groupOwned or groupOwner??

File: controllers/userCtrl.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        whereObj["is_active"]=body.is_active
+    }
+    if(body.updatedAt){
+        whereObj['updatedAt']={contains:body.updatedAt}
+    }
+    if(body.lastLogin){
+        whereObj['lastLogin']={contains:body.lastLogin}
+    }
+    return whereObj;
+}
```
Comment (rishibroto-Tectoro) [inline controllers/userCtrl.ts:49]: Not required. Do the filtering in the where clause.

File: controllers/userCtrl.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+export const updateGroup = async (req: Request, res: Response) => {
+    try {
+        
+        const obj=updateData(req.body);
```
Comment (rishibroto-Tectoro) [inline controllers/userCtrl.ts:54]: Not required. Handle it in where clause in prisma query

File: controllers/userCtrl.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        console.log(obj)
+        const data = await prisma.group.update({where:{id:req.body.id},data:obj})
```
Comment (rishibroto-Tectoro) [inline controllers/userCtrl.ts:56]: Why check with id only?

File: controllers/userCtrl.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    console.log(body)
+    let whereObj:any={}
+    if(body.name){
+        whereObj["name"]=body.name
+    }
+    if(body.hasOwnProperty('is_active')){
+        whereObj["is_active"]=body.is_active
+    }
+    return whereObj;
+}
```
Comment (rishibroto-Tectoro) [inline controllers/userCtrl.ts:73]: Not required. Handle it in where clause or Prisma query.

**Review by rishibroto-Tectoro — CHANGES_REQUESTED** [review]

File: controllers/userCtrl.ts (hunk 2 above)
Comment (Salman-kutty) [reply controllers/userCtrl.ts]: Yes it is groupOwned .

Comment (Salman-kutty) [reply controllers/userCtrl.ts]: Ok corrected.

File: controllers/userCtrl.ts (hunk 3 above)
Comment (Salman-kutty) [reply controllers/userCtrl.ts]: Ok corrected.

File: controllers/userCtrl.ts (hunk 4 above)
Comment (Salman-kutty) [reply controllers/userCtrl.ts]: Ok fixed.

File: controllers/userCtrl.ts (hunk 5 above)
Comment (Salman-kutty) [reply controllers/userCtrl.ts]: Ok fixed.

File: controllers/userCtrl.ts (hunk 6 above)
Comment (Salman-kutty) [reply controllers/userCtrl.ts]: Ok fixed.

**Review by rishibroto-Tectoro — APPROVED** [review]

**Pull request merged**